crates/wasi-common/src/sys/windows/host_impl.rs

```diff
@@ -40,45 +40,70 @@ pub(crate) fn errno_from_win(error: winx::winerror::WinError) -> wasi::__wasi_er
     }
 }
 
-pub(crate) fn fdflags_from_win(mode: AccessMode) -> wasi::__wasi_fdflags_t {
+pub(crate) fn fdflags_from_win(
+    access: AccessMode,
+    mode: FileModeInformation,
+) -> wasi::__wasi_fdflags_t {
     let mut fdflags = 0;
-    // TODO verify this!
-    if mode.contains(AccessMode::FILE_APPEND_DATA) {
+
+    if access.contains(AccessMode::FILE_APPEND_DATA)
+        && !access.contains(AccessMode::FILE_WRITE_DATA)
+    {
         fdflags |= wasi::__WASI_FDFLAG_APPEND;
     }
-    if mode.contains(AccessMode::SYNCHRONIZE) {
-        fdflags |= wasi::__WASI_FDFLAG_DSYNC;
-        fdflags |= wasi::__WASI_FDFLAG_RSYNC;
+
+    if mode.contains(FileModeInformation::FILE_WRITE_THROUGH) {
+        // Only report __WASI_FDFLAG_SYNC, which is technically the only one of the O_?SYNC flags Windows supports.
```

Comment:
I believe before 2.6.33, the all the flags were defined as `O_SYNC`, which is effectively what this is doing here.

I'm open to returning `ENOTSUP` for `O_DSYNC` and `O_RSYNC` if we feel that's necessary.

Reply:
Precisely, on old kernels the flags were all the same, currently glibc defines `O_RSYNC` as `O_SYNC` (which is semantially incorrect, but well). [FreeBSD doesn't support those identifiers at all](https://github.com/freebsd/freebsd/blob/master/sys/sys/fcntl.h) but kernels usually ignore unknown flags for future compatibility. Therefore I think we'd rather either (a) make those three synonymous or (b) ignore `O_RSYNC` and `O_DSYNC`. Personally, I'm rather in favor of (a) because if the programmer requires extra synchronization, it's better to err on the safe side.

Reply:
I'm now treating the three flags as the same (as if the latter two were `#define`'d to `O_SYNC`).

`fd_fdstat_get` only returns `__WASI_FDFLAG_SYNC` as we're treating all three flags as the same.